equinix/resource_metal_device.go

```diff
@@ -726,33 +682,31 @@ func resourceMetalDeviceRead(ctx context.Context, d *schema.ResourceData, meta i
 }
 
 func resourceMetalDeviceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	meta.(*config.Config).AddModuleToMetalUserAgent(d)
-	client := meta.(*config.Config).Metal
+	meta.(*config.Config).AddModuleToMetalGoUserAgent(d)
+	client := meta.(*config.Config).Metalgo
+
+	ur := metalv1.DeviceUpdateInput{}
 
+	// TODO does this need to be a separate device update request?
```

Comment:
@displague forgot about this question:  the packngo version of this resource handled `locked` changes by running a separate update request that only changes the `locked` property of the device and then running a second update request to change the remaining properties.

I didn't find anything in the docs to indicate that `locked` has to be updated on its own; do we still need to update the `locked` property in a separate request from the other properties?

Reply:
We reached out to the API team for help understanding the behavior around `locked`; waiting for confirmation that we don't need a separate request to update that field before touching others.  Once we know the behavior we can also document it in the resource docs.